integrations/discord/events.go

```diff
@@ -7,19 +7,51 @@ import (
 	"go.uber.org/zap"
 
 	"go.autokitteh.dev/autokitteh/integrations/discord/internal/vars"
+	"go.autokitteh.dev/autokitteh/sdk/sdktypes"
 )
 
 func (h *handler) handleEvent(event any, eventType string) {
-	akEvent, err := h.transformEvent(event, eventType)
-	if err != nil {
+	var authorID string
+
+	// TODO(ENG-1546) - Add support for more event types
+	switch e := event.(type) {
+	case *discordgo.MessageCreate:
+		authorID = e.Author.ID
+	case *discordgo.MessageUpdate:
+		authorID = e.Author.ID
+	case *discordgo.MessageDelete:
+		authorID = e.Author.ID
+	default:
+		h.logger.Error("Unsupported event type", zap.String("eventType", eventType))
 		return
 	}
+
 	cids, err := h.vars.FindConnectionIDs(context.Background(), h.integrationID, vars.BotToken, "")
 	if err != nil {
 		h.logger.Error("Failed to find connection IDs", zap.Error(err))
 		return
 	}
-	h.dispatchAsyncEventsToConnections(cids, akEvent)
+	akEvent, err := h.transformEvent(event, eventType)
+	if err != nil {
+		return
+	}
+
+	// Filter out connections from the application itself
+	var validCIDs []sdktypes.ConnectionID
+	for _, cid := range cids {
```

Comment:
You can use a `kittehs`-based transform to do this, but it doesn't really make the code better. So just FYI.